### DISCUSSION TITLE
Authenticated but can't fetch or push

### DISCUSSION THREAD

**Discussion by tinne26** [community/community#27456] [New to GitHub] [answered]
I've done this in the past on Linux and haven't had any issues, but I'm struggling to push to a repository on github from Windows 11. The issue I'm experiencing is very similar to <https://github.com/orgs/community/discussions/23059>, but I don't have any deploy keys (I'm testing on a new private repository):
```
C:\Users\tinne26\dev\tests\hello>git push -u origin main 
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

C:\Users\tinne26\dev\tests\hello>git remote -v
origin  git@github.com:tinne26/helloworld.git (fetch)
origin  git@github.com:tinne26/helloworld.git (push)

C:\Users\tinne26\dev\tests\hello>ssh -T git@github.com
Hi tinne26! You've successfully authenticated, but GitHub does not provide shell access.
```

The only thing I could think of is an incorrect remote address. But I double checked, triple checked, copied directly from the github page, made a new repository with a different name and tried again, and it doesn't work.

I repeated the process in different orders, tried with fetch instead of push, etc. I keep hitting the same wall, and it's driving me insane. Is there any other requirement that I'm missing and could possibly be wrong?

Edit: I've also gone through everything in <https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey>, everything is in order.

**Comment by airtower-luna** [comment, marked as answer]
I've seen similar things happen to people who use Windows, where somehow `git` and `ssh` had different ideas about what is the user's home directory, and thus from where to load keys. :sweat_smile: 

Things to try:

* Try pushing from Git Bash instead of Powershell/Cmd
* Set [`GIT_SSH_COMMAND`](https://git-scm.com/docs/git#Documentation/git.txt-codeGITSSHCOMMANDcode) to `ssh -v` to get debug info as Git is using SSH. This should tell you where it's trying to find keys.

**Reply by tinne26** [reply]
Thank you so so much. On running git again, now I could see that indeed powershell is being dumb (I tried both with cmd and powershell, and none of them worked...):
```
debug1: Will attempt key: /c/Users/tinne26/.ssh/id_rsa
```
... No additional words needed.

**Reply by tinne26** [reply]
Just for context, I think the problem comes from the bundled ssh version in git-scm being an incorrect one (like a Unix version compiled for Windows). Reinstalling git-scm without the bundled ssh was the only clean fix for the issue. I brought that up with git-scm on an existing issue, so we will see soon, but mentioning it here in case it helps other people.

Update: turns out the bundled ssh is for MSYS, and the /c/ paths are actually "normal" in that context (*not like I managed to make it work even under that assumption*). If you want to use cmd / powershell, probably best to stick to the default Windows ssh release and not install the bundled ssh in git-scm (which is the default option).

**Reply by syamsoul** [reply]
> Just for context, I think the problem comes from the bundled ssh version in git-scm being an incorrect one (like a Unix version compiled for Windows). Reinstalling git-scm without the bundled ssh was the only clean fix for the issue. I brought that up with git-scm on an existing issue, so we will see soon, but mentioning it here in case it helps other people.
> 
> Update: turns out the bundled ssh is for MSYS, and the /c/ paths are actually "normal" in that context (_not like I managed to make it work even under that assumption_). If you want to use cmd / powershell, probably best to stick to the default Windows ssh release and not install the bundled ssh in git-scm (which is the default option).

thank you.. it is working for me... using windows openssh... and re-install the git bash without ssh-bundled... and restart ...and it's working

**Reply by Kokodod** [reply]
This solved it for me too, it was doing my head in...

**Reply by Pierre-Sassoulas** [reply]
The ``export GIT_SSH_COMMAND="ssh -v"`` helped tremendously, I could then compare the output for the working ``ssh -vT git@github.com
`` and ``git push`` and realize that:
```diff
-debug1: Authenticating to github.com:22 as '\302\226git'
+debug1: Authenticating to github.com:22 as 'git'

```

**Reply by JackieHanLab** [reply]
Thank you so much, this finally solved my situation!

**Reply by Km1l0** [reply]
I just had to get here and say thank you. I was starting to get crazy with this damn thing, nothing seemed to work but I reinstalled and I was then able to make the push.

**Comment by AkashicSeer** [comment]
OK anyone have the answer here? I knew programming on Windows 10 was a horrible idea, but I didn't expect this BS. I at least thought I could push to a damn repository on github FFS. I installed git for windows. Is there any solution to this other than switching back to Linux so it "just works"
